src/include/txn.i

```diff
@@ -532,17 +534,15 @@ static inline int
 __wt_txn_idle_cache_check(WT_SESSION_IMPL *session)
 {
 	WT_TXN *txn;
-	WT_TXN_STATE *txn_state;
 
 	txn = &session->txn;
-	txn_state = WT_SESSION_TXN_STATE(session);
 
 	/*
 	 * Check the published snap_min because read-uncommitted never sets
 	 * WT_TXN_HAS_SNAPSHOT.
 	 */
 	if (F_ISSET(txn, WT_TXN_RUNNING) &&
-	    !F_ISSET(txn, WT_TXN_HAS_ID) && txn_state->pinned_id == WT_TXN_NONE)
+	    !F_ISSET(txn, WT_TXN_PUBLIC_ID | WT_TXN_PUBLIC_PINNED_ID))
```

Comment:
I don't think this is right. My understanding is that this will return true if WT_TXN_PUBLIC_ID and/or WT_TXN_PUBLIC_PINNED_ID is set. The previous test was ensuring they are both set.